spec/classes/mod/passenger_spec.rb

```diff
@@ -2,6 +2,185 @@
+        'passenger_spawn_method' => {Type: ["smart", "direct"], PassOpt: :PassengerSpawnMethod},
+        'passenger_start_timeout' => {Type: 'Integer', PassOpt: :PassengerStartTimeout},
+        'passenger_startup_file' => {Type: 'RelPath', PassOpt: :PassengerStartupFile},
+        'passenger_stat_throttle_rate' => {Type: 'Integer', PassOpt: :PassengerStatThrottleRate},
+        'passenger_sticky_sessions' => {Type: 'OnOff', PassOpt: :PassengerStickySessions},
+        'passenger_sticky_sessions_cookie_name' => {Type: 'String', PassOpt: :PassengerStickySessionsCookieName},
+        'passenger_thread_count' => {Type: 'Integer', PassOpt: :PassengerThreadCount},
+        'passenger_use_global_queue' => {Type: 'String', PassOpt: :PassengerUseGlobalQueue},
+        'passenger_user' => {Type: 'String', PassOpt: :PassengerUser},
+        'passenger_user_switching' => {Type: 'OnOff', PassOpt: :PassengerUserSwitching},
+        'rack_auto_detect' => {Type: 'String', PassOpt: :RackAutoDetect},
+        'rack_base_uri' => {Type: 'String', PassOpt: :RackBaseURI},
+        'rack_env' => {Type: 'String', PassOpt: :RackEnv},
+        'rails_allow_mod_rewrite' => {Type: 'String', PassOpt: :RailsAllowModRewrite},
+        'rails_app_spawner_idle_time' => {Type: 'String', PassOpt: :RailsAppSpawnerIdleTime},
+        'rails_auto_detect' => {Type: 'String', PassOpt: :RailsAutoDetect},
+        'rails_base_uri' => {Type: 'String', PassOpt: :RailsBaseURI},
+        'rails_default_user' => {Type: 'String', PassOpt: :RailsDefaultUser},
+        'rails_env' => {Type: 'String', PassOpt: :RailsEnv},
+        'rails_framework_spawner_idle_time' => {Type: 'String', PassOpt: :RailsFrameworkSpawnerIdleTime},
+        'rails_ruby' => {Type: 'String', PassOpt: :RailsRuby},
+        'rails_spawn_method' => {Type: 'String', PassOpt: :RailsSpawnMethod},
+        'rails_user_switching' => {Type: 'String', PassOpt: :RailsUserSwitching},
+        'union_station_filter' => {Type: 'QuotedString', PassOpt: :UnionStationFilter},
+        'union_station_gateway_address' => {Type: 'URI', PassOpt: :UnionStationGatewayAddress},
+        'union_station_gateway_cert' => {Type: 'FullPath', PassOpt: :UnionStationGatewayCert},
+        'union_station_gateway_port' => {Type: 'Integer', PassOpt: :UnionStationGatewayPort},
+        'union_station_key' => {Type: 'String', PassOpt: :UnionStationKey},
+        'union_station_proxy_address' => {Type: 'URI', PassOpt: :UnionStationProxyAddress},
+        'union_station_support' => {Type: 'OnOff', PassOpt: :UnionStationSupport},
+        'wsgi_auto_detect' => {Type: 'String', PassOpt: :WsgiAutoDetect},
+        'rails_autodetect' => {Type: 'OnOff', PassOpt: :RailsAutoDetect},
+        'rack_autodetect' => {Type: 'OnOff', PassOpt: :RackAutoDetect},
+    }
+    passenger_config_options.each do |config_option, config_hash|
+      puppetized_config_option = config_option
+      valid_config_values = []
+      case config_hash[:Type]
```

Comment:
I'm curious about the sorting of these tests into cases by the `Type`. Is this just a semantic thing to make the test cases more organized or does it serve a functional purpose?